packages/gatsby-theme-store/src/sdk/img/fileManager.ts

```diff
@@ -10,11 +10,17 @@ interface Options {
   quality?: number
 }
 
-export const optimize = (src: string, opts: Options) => {
+export const optimize = (src: string, opts: Options, ratio = 1) => {
   const url = new URL(src)
 
   for (const option of Object.keys(opts)) {
-    url.searchParams.append(option, `${opts[option as keyof Options]}`)
+    let paramValue = opts[option as keyof Options]
+
+    if (paramValue && ['width', 'height'].includes(option)) {
+      paramValue = (paramValue as number) * ratio
+    }
+
+    url.searchParams.append(option, `${paramValue}`)
```

Comment:
I tried to change to `url.searchParams.append(option, paramValue)` but I think we can only force a string there. 🤔